lxd/api.go

```diff
@@ -325,7 +325,7 @@ func setCORSHeaders(rw http.ResponseWriter, req *http.Request, config *clusterCo
 // notifying us of some user-initiated API request that needs some action to be
 // taken on this node as well.
 func isClusterNotification(r *http.Request) bool {
-	return r.Header.Get("User-Agent") == request.UserAgentNotifier
+	return r != nil && r.Header.Get("User-Agent") == request.UserAgentNotifier
```

Comment:
Can you move this check into `updateClusterCertificate` as its quite specialized, rather than having it here which is quite confusing seeing as its a specific argument to this function.